org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventConditions.java

```diff
@@ -0,0 +1,335 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *******************************************************************************/
+
+package org.osgi.test.assertj.frameworkevent;
+
+import static org.assertj.core.api.Assertions.allOf;
+import static org.assertj.core.api.Assertions.not;
+
+import org.assertj.core.api.Assertions;
+import org.assertj.core.api.Condition;
+import org.assertj.core.api.ListAssert;
+import org.assertj.core.api.ObjectAssert;
+import org.assertj.core.condition.MappedCondition;
+import org.assertj.core.condition.VerboseCondition;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleEvent;
+import org.osgi.framework.FrameworkEvent;
+import org.osgi.test.assertj.bundle.BundleConditions;
+import org.osgi.test.common.bitmaps.Bitmap;
+import org.osgi.test.common.bitmaps.FrameworkEventType;
+
+/**
+ * A Utility-Class thats Provides public static methods to create
+ * {@link FrameworkEventConditions} for an {@link FrameworkEvent}
+ */
+public final class FrameworkEventConditions {
+
+	/**
+	 * Creates a {@link Condition} to be met by an {@link FrameworkEvent}.
+	 * Checking if a {@link Bundle} is <b>equal</b> the bundle that is
+	 * associated with the event {@link BundleEvent}.
+	 * <p>
+	 * Example:
+	 *
+	 * <pre>
+	 * List<FrameworkEvent> frameworkEvents = null;
+	 *
+	 * public static void example_bundleEquals(Bundle bundle) {
+	 *
+	 * 	assertThat(bundleEvents)// created an {@link ListAssert}
+	 * 		.have(bundleEquals(bundle))
+	 * 		.filteredOn(bundleEquals(bundle))
+	 * 		.first()// map to {@link ObjectAssert}
+	 * 		.has(bundleEquals(bundle));// used on {@link ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @param bundle - the bundle that would be checked against the bundle of
+	 *            the {@link FrameworkEvent}
+	 * @return the Condition<br>
+	 */
+	public static Condition<FrameworkEvent> bundleEquals(Bundle bundle) {
+
+		return MappedCondition.mappedCondition(FrameworkEvent::getBundle, BundleConditions.sameAs(bundle),
+			"FrameworkEvent::getBundle");
+
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an {@link FrameworkEvent}.
+	 * Checking if the bundle that is associated with the eventof the
+	 * {@link FrameworkEvent} <b>is not null</b>.
+	 *
+	 * <pre>
+	 * List<FrameworkEvent> frameworkEvents = null;
+	 *
+	 * public static void example_bundleIsNotNull() {
+	 *
+	 * 	assertThat(frameworkEvents)// created an {@link ListAssert}
+	 * 		.have(bundleIsNotNull())
+	 * 		.filteredOn(bundleIsNotNull())
+	 * 		.first()// map to {@link ObjectAssert}
+	 * 		.has(bundleIsNotNull());// used on {@link ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @return the Condition
+	 */
+	public static Condition<FrameworkEvent> bundleIsNotNull() {
```

Comment:
Do we need to make both `is` and `isNot` versions of different conditions? People can use the `Assertions.not` method to do `not(bundleIsNull())`. This also reduces the API size and testing complexity.